### PR TITLE
[Broadcast] migrate all subscriptions to broadcaster-based subscription

### PR DESCRIPTION
**Pull request by bariqhibat**
Signed-off-by: Bariq <bariqhibat@gmail.com>

**Description**

Fixes https://github.com/meshery/meshery/issues/3712

Migrate subscription for dataplane to broadcast-based subscription.

**Notes for Reviewers**
Turns out I don't need to add a type in the Meshkit.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — DISMISSED** [review]

File: internal/graphql/resolver/data-plane.go [hunk 1]
```diff
@@ -41,18 +43,23 @@ func (r *Resolver) listenToDataPlaneState(ctx context.Context, provider models.P
 
 		for {
 			select {
-			case <-r.MeshSyncChannel:
+			case <-dataPlaneSyncChannel:
+				r.Log.Info("Dataplane sync channel called")
```
Comment (dhruv0000) [inline internal/graphql/resolver/data-plane.go:47]: Need to do a type check here, or else it will be triggered for other types too.

Comment (dhruv0000) [reply internal/graphql/resolver/data-plane.go:47]: We were using `MeshSyncChannel` for trigger, right. We should use it here too.

Also, you might need to update other places too that uses the `MeshSyncChannel` channel to use broadcast instead. (including handler ([here](https://github.com/meshery/meshery/blob/master/handlers/handler_instance.go#L29)) and Meshery-extension ([here](https://github.com/meshery/meshery/blob/de283e40e3b1477c79e64fdac7a4d1b7de8ad082/handlers/extensions.go#L44)) too).

**Review by dhruv0000 — CHANGES_REQUESTED** [review]

File: internal/graphql/resolver/data-plane.go (hunk 1 above)
Comment (bariqhibat) [reply internal/graphql/resolver/data-plane.go:47]: Yeah, do we want to move this meshsync channel to the broadcaster? If so, there'll be a lot of changes for this @dhruv0000

**Comment by bariqhibat** [comment]
I'll migrate the rest of the subscriptions in this PR. And will make another PR for migrating MeshSyncChannel to broadcast-based system

**Comment by bariqhibat** [comment]
Hey @leecalcote, I might need to do a change in the meshkit for the MeshSyncChannel that are not found in the enums. 

This might be an issue in the future as I might need to do a lot of small changes like this, should I just convert the type to be string instead of type `BroadcastSource`? 
Ref: https://github.com/layer5io/meshkit/blob/master/utils/broadcast/broadcaster.go#L26

// @dhruv0000

**Comment by bariqhibat** [comment]
Well for now, I'll just use the Source of `OperatorSyncChannel` and filter the broadcast message according to Type instead.

**Comment by leecalcote** [comment]
Feedback @dhruv0000?

**Comment by dhruv0000** [comment]
> Well for now, I'll just use the Source of `OperatorSyncChannel` and filter the broadcast message according to Type instead.

We can also define the `BroadcastSource type` in Meshery repo itself too if that makes thing easier (package like `helper` might be a good place).

Using Type currently will just increase your work later on. in my opinion. :sweat_smile:

**Comment by dhruv0000** [comment]
> I'll migrate the rest of the subscriptions in this PR. And will make another PR for migrating MeshSyncChannel to broadcast-based system

Sure. Makes sense, since it would involve migrating meshery-extensions to broadcaster too.

**Comment by dhruv0000** [comment]
Great work btw @bariqhibat :fist_raised: .

**Comment by bariqhibat** [comment]
Hey @utkarsh-pro, could you help reviewing this as well? If possible, this [PR](https://github.com/layer5labs/meshery-extensions/pull/355) as well, since it is dependent to this. If we're good to go, we'll merge them both

**Pull request closed without merging**